Source/com/IUnknown.h

```diff
@@ -334,6 +334,9 @@ namespace ProxyStub {
 
             _adminLock.Unlock();
 
+            if (result != Core::ERROR_NONE) {
+                delete &_parent;
```

Comment:
Couldn't we better make _parent a pointer instead of a reference? Makes the ownership more clear and then we could also set it to nullptr here

Reply:
Nope, This is a linked through reference counted object (composit of the parent). If it deletes the parent it deletes this one as well. Lifetime between the two objects (as this is a composit of the parent) are equal. Do not want to rasie the suggestion that it is an aggregate or that the two can live without each other. So the reference has exactly the same lifetime as the parent (you can not event construct this object without the parent)

Reply:
Clear!